pyproject.toml

```diff
@@ -0,0 +1,2 @@
+[build-system]
+requires=["setuptools", "cython", "wheel"]
```

Comment:
There seem to be various unrelated changes bundled in here. This one is possibly what breaks building the wheels in CI.

Reply:
It is better to keep different changes like this in separate pull requests so this one can be for adding `fmpz_mpoly` but changes for flint 3 would go somewhere else (actually we already have a gh-43 for that).

Any changes to the build system should definitely be kept separate from anything else so that we can see clearly the impact of each change independently.

Reply:
Thanks for the info on a coefficient/exponent dict.
Sorry about the build system confusion (I forgot that I did that I'll see about backing it out)
I agree about a common base class (one exists, but it is currently empty)

Reply:
There are actually three unrelated changes, include changes for flint 3.0, build system changes, and fmpz_changes.  Sorry, I'll get to it.

Reply:
Ok, I've backed out the flint-3 and setuptools and will see if that makes the CI happy.  I am building a copy of flint-2.9 so I can test it.  What is the version of arb that corresponds to flint-2.9?